git-commit-signing/run.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env sh
+
+if ! command -v git > /dev/null; then
+        echo "git is not installed"
+        exit 1
+fi
+
+if ! command -v jq > /dev/null; then
+        echo "jq is not installed"
+        exit 1
+fi
+
+mkdir -p ~/.ssh
+
+echo "Downloading SSH key"
+
+ssh_key=$(curl --request GET \
+        --url "${CODER_AGENT_URL}api/v2/workspaceagents/me/gitsshkey" \
+        --header "Coder-Session-Token: ${CODER_AGENT_TOKEN}")
+
+jq --raw-output ".public_key" > ~/.ssh/coder.pub <<EOF
+$ssh_key
+EOF
+
+jq --raw-output ".private_key" > ~/.ssh/coder <<EOF
+$ssh_key
+EOF
+
+chmod -R 400 ~/.ssh/coder
+chmod -R 400 ~/.ssh/coder.pub
+
+echo "Configuring git to use the SSH key"
+
+git config --global gpg.format ssh
+git config --global commit.gpgsign true
+git config --global user.signingkey ~/.ssh/coder
```

Comment:
I guess we should signal it in the README maybe, but isn't that the case with the `git-config` module too ?

Reply:
Yes, good observation. It's absolutely (going to be) a problem in other modules as well, and `git-config` exhibits exactly the same problem as I described above.

Reply:
6d747f3416aa2dcf0241b1ec9979405710b55f0d

---

@mafredri what do you think ? I will also create a separate PR for `git-config` once you approve the wording.

Reply:
I think it looks good. I'd suggest a few more changes so that it flows better when reading:

> This module may conflict with the users dotfiles, or modules like the personalize-module. A conflict may happen if they replace the git config file or have configuration directives that overwrite the ones in this modules.